src/Cake.Core/Scripting/ScriptRunner.cs

```diff
@@ -99,13 +99,11 @@ public void Run(IScriptHost host, FilePath scriptPath, IDictionary<string, strin
                 throw new ArgumentNullException("arguments");
             }
 
-            // Prepare the context.
-            host.Context.Arguments.SetArguments(arguments);
-            host.Context.Environment.WorkingDirectory = scriptPath.MakeAbsolute(host.Context.Environment).GetDirectory();
+            // Prepare the environment.
+            _environment.WorkingDirectory = scriptPath.MakeAbsolute(_environment).GetDirectory();
 
             // Analyze the script file.
             _log.Verbose("Analyzing build script...");
-            scriptPath = scriptPath.IsRelative ? scriptPath.MakeAbsolute(_environment) : scriptPath;
```

Comment:
Note to self, correct, this is already done in:
https://github.com/cake-build/cake/blob/c3f1280b6edc6a4983ccea655cca3e5a8efab805/src/Cake.Core/Scripting/Analysis/ScriptAnalyzer.cs#L75